src/subplots.ts

```diff
@@ -1358,10 +1477,523 @@ export class PrimitiveGroupContainer extends PlotData {
+
+  set nYTicks(value: number) {this._nYTicks = value}
+
+  private buildNumberAxis(frameOrigin: Vertex, yEnd: Vertex): newAxis {
+    const numberAxis = this.setAxis('number', frameOrigin, yEnd, this.nYTicks);
+    numberAxis.minValue = 0;
+    numberAxis.maxValue = Math.max(...this.features.get(this.yFeature)) + 1;
+    numberAxis.nTicks = this.nYTicks;
+    numberAxis.saveLoc();
+    return numberAxis
+  }
+
+  private updateNumberAxis(numberAxis: newAxis, bars: Bar[]): newAxis {
+    this.features.set('number', this.getNumberFeature(bars));
+    numberAxis.maxValue = Math.max(...this.features.get(this.yFeature)) + 1;
+    numberAxis.saveLoc();
+    return numberAxis
+  }
+
+  private getNumberFeature(bars: Bar[]): number[] {
+    const numberFeature = Array.from(Array(this.features.get(this.xFeature).length), () => 0);
+    bars.forEach(bar => {bar.values.forEach(value => numberFeature[value] = bar.length)});
+    return numberFeature
+  }
+
+  private boundedTicks(axis: newAxis): number[] {
+    let fakeTicks = [axis.minValue].concat(axis.ticks);
+    fakeTicks.push(axis.maxValue)
+    return fakeTicks
+  }
+
+  private storeBarState(): [number[], number[]] {
+    const [hovered, clicked] = [[] as number[], [] as number[]];
+    if (this.bars) {
+      this.bars.forEach(bar => {
+        if (bar.isHover) {hovered.push(...bar.values)};
+        if (bar.isClicked) {clicked.push(...bar.values)};
+      })
+    }
+    return [hovered, clicked]
+  }
+
+  private computeBars(axis: newAxis, vector: number[]): Bar[] {
+    const numericVector = axis.stringsToValues(vector);
+    [this.hoveredIndex, this.clickedIndex] = this.storeBarState();
+    let fakeTicks = this.boundedTicks(axis);
+    let bars = Array.from(Array(fakeTicks.length - 1), () => new Bar());
+    numericVector.forEach((value, valIdx) => {
+      for (let tickIdx = 0 ; tickIdx < fakeTicks.length - 1 ; tickIdx++ ) {
+        if (value >= fakeTicks[tickIdx] && value < fakeTicks[tickIdx + 1]) {
+          bars[tickIdx].values.push(valIdx);
+          break
+        }
+      }
+    });
+    return bars
+  }
+
+  public draw(): void {
+    this.updateAxes();
+    this.getBarsDrawing();
+
+    this.context_show.setTransform(this.movingMatrix);
+    this.bars.forEach(bar => {bar.buildPath() ; bar.draw(this.context_show)});
+    this.context_show.resetTransform();
+    this.movingObjects = this.bars;
+
+    this.context_show.setTransform(this.canvasMatrix);
+    super.drawAxes();
+    this.context_show.resetTransform();
+  }
+
+  public updateAxes(): void {
+    super.drawCanvas();
+    this.context_show.setTransform(this.canvasMatrix);
+    super.updateAxes()
+    this.bars = this.computeBars(this.axes[0], this.features.get(this.xFeature));
+    this.axes[1] = this.updateNumberAxis(this.axes[1], this.bars);
+  }
+
+  public getBarsDrawing() {
+    const fullTicks = this.boundedTicks(this.axes[0]);
+    for (let barIdx = 0 ; barIdx < this.bars.length ; barIdx++) {
```

Comment:
As you use this.bars[barIdx] after this, should you use forEach ?

Reply:
Done